promql/functions.go

```diff
@@ -716,16 +716,29 @@ func linearRegression(samples []Point, interceptTime int64) (slope, intercept fl
 		sumY, cY   float64
 		sumXY, cXY float64
 		sumX2, cX2 float64
+		initY      float64
+		constY     bool
 	)
-	for _, sample := range samples {
+	initY = samples[0].V
+	constY = true
+	for i, sample := range samples {
+		// Reset constY to false if any new y values are encountered
+		if constY && i > 0 && sample.V != initY {
+			constY = false
+		}
 		n += 1.0
 		x := float64(sample.T-interceptTime) / 1e3
 		sumX, cX = kahanSumInc(x, sumX, cX)
 		sumY, cY = kahanSumInc(sample.V, sumY, cY)
 		sumXY, cXY = kahanSumInc(x*sample.V, sumXY, cXY)
 		sumX2, cX2 = kahanSumInc(x*x, sumX2, cX2)
 	}
-
+	if constY {
+		if math.IsInf(initY, 0) {
+			initY = math.NaN()
```

Comment:
I think the outcome of the discussion was to return `return math.NaN(), math.NaN()`. This matches the current behavior which returns NaN rather than 0 for the slope of a time series consisting entirely of +Inf values.

Reply:
Ah you’re right, I’ll fix that.